app/src/main/java/com/kickstarter/viewmodels/MessageThreadHolderViewModel.java

```diff
@@ -88,7 +99,36 @@ public ViewModel(final @NonNull Environment environment) {
       this.unreadCountTextViewText = this.messageThread
         .map(MessageThread::unreadMessagesCount)
         .map(NumberUtils::format);
-      this.unreadIndicatorViewHidden = this.messageThread.map(m -> m.unreadMessagesCount() == 0);
+
+      this.unreadIndicatorViewHidden = hasUnreadMessages.map(BooleanUtils::negate);
+
+      this.messageThread
+        .compose(takeWhen(this.messageThreadCardViewClicked))
+        .subscribe(thread -> markedAsRead(thread, this.sharedPreferences));
+    }
+
+    private static @NonNull String cacheKey(final @NonNull MessageThread messageThread) {
+      return SharedPreferenceKey.MESSAGE_THREAD_HAS_UNREAD_MESSAGES + messageThread.id();
+    }
+
+    private static boolean hasUnreadMessages(final @NonNull MessageThread messageThread,
+      final @NonNull SharedPreferences sharedPreferences) {
+      return sharedPreferences.getBoolean(cacheKey(messageThread), messageThread.unreadMessagesCount() > 0);
+    }
+
+    private static void markedAsRead(final @NonNull MessageThread messageThread,
+      final @NonNull SharedPreferences sharedPreferences) {
+      final SharedPreferences.Editor editor = sharedPreferences.edit();
+      editor.putBoolean(cacheKey(messageThread), false);
+      editor.apply();
+    }
+
+    private static void setHasUnreadMessagesPreference(final @NonNull MessageThread messageThread,
+      final @NonNull SharedPreferences sharedPreferences) {
+
+      final SharedPreferences.Editor editor = sharedPreferences.edit();
+      editor.putBoolean(cacheKey(messageThread), messageThread.unreadMessagesCount() > 0);
+      editor.apply();
```

Comment:
actually, scratch that for now. seems like we don't need to set this in the AppModule for message thread's unread messages, since it's a preference specific to each message thread id rather than a global preference. 

but this is still a valuable solution for setting a `hasUnreadMessages` boolean preference for the user that may avoid having to refresh the user onResume each  time so will explore that